File: testJava/src/test/java/E2E/poms/simpleInput/simpleInputPOM.java
package E2E.poms.simpleInput;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class simpleInputPOM {

    private WebDriver webDriver;

    public simpleInputPOM(WebDriver webDriver) {
        this.webDriver = webDriver;

        PageFactory.initElements(webDriver, this);
    }

    // ------------------- Single Input Field --------------------------------

    @FindBy(id = "user-message")
    public WebElement messageInput;

    @FindBy(xpath = "//*[@id=\"get-input\"]/button")
    public WebElement showMessageButton;

    @FindBy(id = "display")
    public WebElement displayMessageSpan;

    // ------------------- Two Input Fields --------------------------------

    @FindBy(id = "sum1")
    public WebElement inputA;

    @FindBy(id = "sum2")
    public WebElement inputB;

    @FindBy(xpath = "//*[@id=\"gettotal\"]/button")
    public WebElement getTotalButton;

    @FindBy(id = "displayvalue")
    public WebElement displayValueSpan;

}

File: seleniumEasyPython/poms/simple_input_POM.py
from selenium.webdriver.chrome.webdriver import WebDriver
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webelement import WebElement


class SimpleInputPOM:
    def __init__(self, driver: WebDriver):
        self.driver = driver

    # pop-up window
    def pop_up_window(self):
        element: WebElement = self.driver.find_element(By.ID, "at-cv-lightbox-close")
        return element

    # ------------------- Single Input Field --------------------------------

    # message input
    def message_input(self):
        element: WebElement = self.driver.find_element(By.ID, "user-message")
        return element

    # show message button
    def show_message_button(self):
        element: WebElement = self.driver.find_element(By.XPATH, '//*[@id="get-input"]/button')
        return element

    # display message
    def display_message_span(self):
        element: WebElement = self.driver.find_element(By.ID, "display")
        return element

    # ------------------- Two Input Fields --------------------------------

    # input A
    def input_a(self):
        element: WebElement = self.driver.find_element(By.ID, "sum1")
        return element

    # input B
    def input_b(self):
        element: WebElement = self.driver.find_element(By.ID, "sum2")
        return element

    # total button
    def show_value_button(self):
        element: WebElement = self.driver.find_element(By.XPATH, '//*[@id="gettotal"]/button')
        return element

    # display value button
    def value_button(self):
        element: WebElement = self.driver.find_element(By.ID, "displayvalue")
        return element
